ts/main.ts

```diff
@@ -28,9 +37,93 @@ document.addEventListener('DOMContentLoaded', () => {
     data.entries.unshift(newEntry);
     data.nextEntryId++;
 
+    const $entry = renderEntry(newEntry);
+    $entriesList.prepend($entry);
+
     $photoPreview.setAttribute('src', 'images/placeholder-image-square.jpg');
     $form.reset();
 
     savedData();
+    toggleNoEntries();
+    viewSwap('entries');
+  });
+
+  data.entries.forEach((entry) => {
+    const $entry = renderEntry(entry);
+    $entriesList.appendChild($entry);
   });
+
+  toggleNoEntries();
+  viewSwap(data.view);
+
+  $newEntryButton.addEventListener('click', (event) => {
+    event.preventDefault();
+    viewSwap('entry-form');
+  });
+
+  if ($entriesLink) {
+    $entriesLink.addEventListener('click', (event) => {
+      event.preventDefault();
+      viewSwap('entries');
+    });
+  }
```

Comment:
Why do you have these in two different ways? One you are not validating to add the event listener but the other you are only adding the event listener if it have a value? Instead you should do what we did before and throw an error if the value is falsy so that you can know if/where this goes wrong.
```
if (!$entriesLink) throw new Error("error querying $entriesLink")

$entriesLink.addEventListener. . .
```